test_architectural_fixes.py

```
@@ -0,0 +1,126 @@
+# Add the package to path
+sys.path.insert(0, '/home/runner/work/PraisonAI/PraisonAI/src/praisonai-agents')
+
+def test_retry_jitter():
+    """Test Gap 2 fix: retry jitter prevents thundering herd"""
+    print("Testing retry jitter fix...")
+    
+    from praisonaiagents.llm.error_classifier import ErrorCategory, get_retry_delay
+    
+    # Test that rate limit delays now have jitter
+    delays = []
+    for i in range(10):
```

Ruff (0.15.11)

[warning] Loop control variable `i` not used within loop body

Rename unused `i` to `_i`

(B007)